src/mlpack/core/data/load_impl.hpp

```diff
@@ -292,130 +296,6 @@ bool Load(const std::string& filename,
   return true;
 }
 
-// For loading data into sparse matrix
-template <typename eT>
-bool Load(const std::string& filename,
-          arma::SpMat<eT>& matrix,
-          const bool fatal,
-          const bool transpose)
-{
-  Timer::Start("loading_data");
-
-  // Get the extension.
-  std::string extension = Extension(filename);
-
-  // Catch nonexistent files by opening the stream ourselves.
-  std::fstream stream;
-#ifdef  _WIN32 // Always open in binary mode on Windows.
-  stream.open(filename.c_str(), std::fstream::in | std::fstream::binary);
-#else
-  stream.open(filename.c_str(), std::fstream::in);
-#endif
-  if (!stream.is_open())
-  {
-    Timer::Stop("loading_data");
-    if (fatal)
-      Log::Fatal << "Cannot open file '" << filename << "'. " << std::endl;
-    else
-      Log::Warn << "Cannot open file '" << filename << "'; load failed."
-          << std::endl;
-
-    return false;
-  }
-
-  bool unknownType = false;
-  arma::file_type loadType;
-  std::string stringType;
-
-  if (extension == "tsv" || extension == "txt")
-  {
-    loadType = arma::coord_ascii;
-    stringType = "Coordinate Formatted Data for Sparse Matrix";
-  }
-  else if (extension == "bin")
-  {
-    // This could be raw binary or Armadillo binary (binary with header).  We
-    // will check to see if it is Armadillo binary.
-    const std::string ARMA_SPM_BIN = "ARMA_SPM_BIN";
-    std::string rawHeader(ARMA_SPM_BIN.length(), '\0');
-
-    std::streampos pos = stream.tellg();
-
-    stream.read(&rawHeader[0], std::streamsize(ARMA_SPM_BIN.length()));
-    stream.clear();
-    stream.seekg(pos); // Reset stream position after peeking.
-
-    if (rawHeader == ARMA_SPM_BIN)
-    {
-      stringType = "Armadillo binary formatted data for sparse matrix";
-      loadType = arma::arma_binary;
-    }
```

Comment:
@heisenbuug it seems that there is a specific header manipulation related to sparse matrices, @conradsnicta @rcurtin Do you think we can remove this one? Or it should stay here?
@heisenbuug Let me know if you bring back this function, would all tests error related to sparse matrices would pass?

Reply:
I don't think just bringing back the function will do.
If you remember our issue was that it was always
entering the `Load()` which was intended for dense
matrix.

With conrad's solution, we handed the `csv` parsing
for the sparse matrix but the logic for all other file types is still
in that `Load()`.

Reply:
> @heisenbuug Let me know if you bring back this function, would all tests error related to sparse matrices would pass?

Even after adding back that function, tests are failing in the same way.

Reply:
> @heisenbuug Let me know if you bring back this function, would all tests error related to sparse matrices would pass?

Even after adding back that function, tests are failing in the same way.